mmcv/ops/csrc/common/cuda/min_area_polygons_cuda.cuh

```diff
@@ -0,0 +1,383 @@
+// Copyright (c) OpenMMLab. All rights reserved
+#ifndef MIN_AREA_POLYGONS_CUDA_KERNEL_CUH
+#define MIN_AREA_POLYGONS_CUDA_KERNEL_CUH
+
+#ifdef MMCV_USE_PARROTS
+#include "parrots_cuda_helper.hpp"
+#else
+#include "pytorch_cuda_helper.hpp"
+#endif
+
+#define MAXN 20
+const float EPS = 1E-8;
+
+int const threadsPerBlock = 512;  // sizeof(unsigned long long) * 8;
+
+__device__ inline int sig(float d) { return int(d > EPS) - int(d < -EPS); }
+
+struct Point {
+  float x, y;
+  __device__ Point() {}
+  __device__ Point(float x, float y) : x(x), y(y) {}
+};
+
+__device__ inline bool point_same(Point &a, Point &b) {
+  return sig(a.x - b.x) == 0 && sig(a.y - b.y) == 0;
+}
+
+__device__ inline void swap1(Point *a, Point *b) {
```

Comment:
Do we have some common implementation of functions like swap for `__device__` in the common folder? @zhouzaida

Reply:
This function is for `Point`, which doesn't have a common implementation in mmcv as far as I know.

Reply:
OK